app/controllers/ConfirmEORIDetailsController.scala

```diff
@@ -0,0 +1,112 @@
+ */
+
+package controllers
+
+import com.google.inject.Inject
+import controllers.actions.{DataRetrievalAction, IdentifierAction}
+import javax.inject.Singleton
+import models.{EoriDetails, UserAnswers}
+import pages.KnownEoriDetails
+import play.api.Logger
+import play.api.i18n.{I18nSupport, Messages}
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
+import repositories.SessionRepository
+import services.EoriDetailsService
+import uk.gov.hmrc.govukfrontend.views.Aliases.SummaryList
+import uk.gov.hmrc.govukfrontend.views.viewmodels.content.{HtmlContent, Text}
+import uk.gov.hmrc.govukfrontend.views.viewmodels.summarylist._
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import views.html.ConfirmEORIDetailsView
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+@Singleton
+class ConfirmEORIDetailsController @Inject()(identify: IdentifierAction,
+                                             getData: DataRetrievalAction,
+                                             mcc: MessagesControllerComponents,
+                                             sessionRepository: SessionRepository,
+                                             eoriDetailsService: EoriDetailsService,
+                                             view: ConfirmEORIDetailsView
+                                            )
+  extends FrontendController(mcc) with I18nSupport {
+
+  private val logger = Logger("application." + getClass.getCanonicalName)
+
+
+  def onLoad(): Action[AnyContent] = (identify andThen getData).async { implicit request =>
+    val userAnswers = request.userAnswers.getOrElse(UserAnswers(request.credId))
+    val getEoriDetails = userAnswers.get(KnownEoriDetails)
+    if (getEoriDetails.isEmpty) {
+      eoriDetailsService.retrieveEoriDetails(request.eori).flatMap {
+        case Right(eoriDetails) =>
+          for {
+            updatedAnswers <- Future.fromTry(userAnswers.set(KnownEoriDetails, eoriDetails))
+            _ <- sessionRepository.set(updatedAnswers)
+          } yield Ok(view(summaryList(eoriDetails).getOrElse(Seq.empty)))
+        case Left(error) =>
+          logger.error(error.message + " " + error.status)
+          Future.successful(NotFound(error.message + " " + error.status))
+      }
+    } else {
+      Future.successful(Ok(view(summaryList(getEoriDetails.get).getOrElse(Seq.empty))))
+    }
+  }
+
+
+  def summaryList(eoriDetails: EoriDetails)(implicit messages: Messages): Option[Seq[SummaryList]] = {
+
+    val eoriNumberSummaryListRow: Option[Seq[SummaryListRow]] = Some(
+      Seq(
+        SummaryListRow(
+          key = Key(
+            content = Text(messages("confirmEORI.eoriNumber")),
+            classes = "govuk-!-width-two-thirds"
+          ),
+          value = Value(
+            content = HtmlContent(eoriDetails.eori)
+          )
+        )
+      ))
+
+
+    val nameSummaryListRow: Option[Seq[SummaryListRow]] = Some(
```

Comment:
eoriNumberSummaryListRow and nameSummaryListRow return almost the same thing, only message is different and content, just refactor this out into a helper method that takes those two values and you'll reduce duplicate code here